tests/tests.h

```diff
@@ -32,6 +32,11 @@
 #include <sstream>
 #include <iomanip>
 
+// silence extra diagnostics in the testsuite
+#ifdef DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#endif
```

Comment:
Shrug. Why would you run the latest testsuite against an old deal.II installation that doesn't export this `#define`? You'd get dozens of failures for other reasons as well.

But you didn't answer why you want to do this in the first place?